db.go

```diff
@@ -278,30 +279,27 @@ func (s *ColumnStore) DB(ctx context.Context, name string) (*DB, error) {
 	// If bucket storage is configured; scan for existing tables in the database
 	if db.bucket != nil {
 		if err := db.bucket.Iter(ctx, "", func(block string) error {
-			attr, err := db.bucket.Attributes(ctx, block)
-			if err != nil {
-				return err
+			if filepath.Base(block) != schemaFileName {
+				return nil
 			}
 
-			// grab table name
-			tableName := filepath.Dir(filepath.Dir(block))
-
-			b := &BucketReaderAt{
-				name:   block,
-				ctx:    ctx,
-				Bucket: db.bucket,
+			r, err := db.bucket.Get(ctx, block)
+			if err != nil {
+				return fmt.Errorf("failed to get schema.json: %w", err)
 			}
+			defer r.Close()
 
-			f, err := parquet.OpenFile(b, attr.Size)
-			if err != nil {
-				return err
+			definition := &schemapb.Schema{}
+			if err := json.NewDecoder(r).Decode(definition); err != nil {
```

Comment:
Should we be using the `jsonpb` encoder/decoder maybe? The advantage we get is that we can have the strinified version of enums that way as opposed to a plain number.

Reply:
yes, I always forget that's a thing